public/app/features/dashboard-scene/conditional-rendering/ConditionalRenderingVariable.tsx

```diff
@@ -56,19 +59,28 @@ export class ConditionalRenderingVariable extends ConditionalRenderingBase<Condi
 
     const variableValue = variable.getValue() ?? '';
 
-    const hit = Array.isArray(variableValue)
-      ? variableValue.includes(this.state.value.value)
-      : variableValue === this.state.value.value;
+    let hit: boolean;
+
+    if (this.state.value.operator === '=' || this.state.value.operator === '!=') {
+      hit = Array.isArray(variableValue)
+        ? variableValue.includes(this.state.value.value.toString())
+        : variableValue === this.state.value.value.toString();
+    } else {
+      const regex = new RegExp(this.state.value.value);
+      hit = Array.isArray(variableValue)
+        ? variableValue.some((currentVariableValue) => regex.test(currentVariableValue.toString()))
+        : regex.test(variableValue.toString());
+    }
```

Comment:
I didn't understand what "added matches/not matches" from the PR description ment until I read this code. Might want to mention regex support in there.

Reply:
Done!